packages/quantic/force-app/main/default/lwc/quanticQueryError/quanticQueryError.js

```diff
@@ -0,0 +1,149 @@
+import { LightningElement, api, track } from 'lwc';
+import {registerComponentForInit, initializeWithHeadless} from 'c/quanticHeadlessLoader';
+import {I18nUtils} from 'c/quanticUtils';
+
+import coveoOnlineHelpLink from '@salesforce/label/c.quantic_CoveoOnlineHelpLink';
+import moreInformation from '@salesforce/label/c.quantic_MoreInformation';
+import checkForMore from '@salesforce/label/c.quantic_CheckForMore';
+import community from '@salesforce/label/c.quantic_Community';
+import contactCoveoSupportTeam from '@salesforce/label/c.quantic_ContactCoveoSupportTeam';
+import goBack from '@salesforce/label/c.quantic_GoBack';
+import disconnectedException from '@salesforce/label/c.quantic_DisconnectedException';
+import noEndpointsException from '@salesforce/label/c.quantic_NoEndpointsException';
+import invalidTokenException from '@salesforce/label/c.quantic_InvalidTokenException';
+import organizationIsPausedException from '@salesforce/label/c.quantic_OrganizationIsPausedException';
+import disconnectedTitle from '@salesforce/label/c.quantic_DisconnectedTitle';
+import noEndpointsTitle from '@salesforce/label/c.quantic_NoEndpointsTitle';
+import invalidTokenTitle from '@salesforce/label/c.quantic_InvalidTokenTitle';
+import organizationIsPausedTitle from '@salesforce/label/c.quantic_OrganizationIsPausedTitle';
+import genericErrorTitle from '@salesforce/label/c.quantic_GenericErrorTitle';
+import disconnectedDesc from '@salesforce/label/c.quantic_DisconnectedDesc';
+import noEndpointsDesc from '@salesforce/label/c.quantic_NoEndpointsDesc';
+import invalidTokenDesc from '@salesforce/label/c.quantic_InvalidTokenDesc';
+import organizationIsPausedDesc from '@salesforce/label/c.quantic_OrganizationIsPausedDesc';
+import genericErrorDesc from '@salesforce/label/c.quantic_GenericErrorDesc';
+
+export default class QuanticQueryError extends LightningElement {
+  /** @type {string} */
+  @api engineId;
+
+  /** @type {import("coveo").QueryError} */
+  queryError;
+  /** @type {import("coveo").Unsubscribe} */
+  unsubscribe;
+
+  /**
+   * @type {string}
+   */
+  @track type;
+  /**
+   * @type {Boolean}
+   */
+  @track hasError;
+  /**
+   * @type {string}
+   */
+  @track error
+
+  showMoreInfo = false;
+
+  labels = {
+    coveoOnlineHelpLink,
+    moreInformation,
+    checkForMore,
+    community,
+    contactCoveoSupportTeam,
+    goBack
+  }
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+  }
+
+  /**
+   * @param {import("coveo").SearchEngine} engine
+   */
+  @api
+  initialize(engine) {
+    this.queryError = CoveoHeadless.buildQueryError(engine);
+    this.unsubscribe = this.queryError.subscribe(() => this.updateState());
+  }
+
+  disconnectedCallback() {
+    if (this.unsubscribe) {
+      this.unsubscribe();
+    }
+  }
+
+  updateState() {
+    this.type = this.queryError.state.error?.type;
```

Comment:
the `?.` is supported now? 😲😲😲😲😲

You used to have to set something in beta mode to use it. It changes EVERYTHING!

Reply:
The quantic code is riddled with long lines to get around this limitation. If true this is going to feel so good.

Reply:
Oh yea!

Reply:
Yes yes !!